index/index_monitor.go

```diff
@@ -0,0 +1,46 @@
+package index
+
+// helpers to atomically switch from using one index to another
+
+import (
+	"sync"
+)
+
+type IndexMonitor struct {
```

Comment:
I like the idea of factoring this out into its own thing, since this functionality seems separate from what the sequins "class" is responsible for. That class already seems to be doing too much, so I'm reluctant to add more. I'm open to finding a different home for this, though.